File: sumitrust2019/b-train.py
def solve(n: int) -> str:
    ng, ok = 0, n
    while ok - ng > 1:
        mid = (ok + ng) // 2
        if int(mid * 1.08) < n:
            ng = mid
        else:
            ok = mid
    return str(ok) if int(ok * 1.08) == n else ":("


def main():
    n = int(input())
    ans = solve(n)
    print(ans)


if __name__ == '__main__':
    main()

File: ABC090/b-train.rs
fn is_palindromic(s: &str) -> bool {
    let s: Vec<_> = s.chars().collect();
    let ln = s.len();
    let mid = ln / 2;
    for i in 0..mid {
        if s[i] != s[ln - i - 1] {
            return false;
        }
    }
    return true;
}

fn main() {
    proconio::input! {
        a: i32, b: i32
    }
    let mut ans = 0;
    for i in a..=b {
        let i_str = i.to_string();
        if is_palindromic(&i_str) {
            ans += 1;
        }
    }
    println!("{}", ans);
}
